crypto/bio/b_sock2.c

```diff
@@ -175,8 +175,10 @@ int BIO_listen(int sock, const BIO_ADDR *addr, int options)
         return 0;
 
 # ifndef OPENSSL_SYS_WINDOWS
-    /* SO_REUSEADDR has different behavior on Windows than on
```

Comment:
On *side* note one can wonder how does it affect Cygwin. I mean Cygwin has no other option by to rely on Windows sockets... So maybe condition has to cover even Cygwin...

Reply:
On related note, since SO_REUSEADDR is special on Windows, maybe it would be appropriate to support its "antipode", SO_EXCLUSIVEADDRUSE. I mean b_sock2.c is supposed to provide OS-neutral interface to sockets, right? So that one write multi-platform application. This kind of implies that it should be expressive enough to do various useful things. And it's not inappropriate to expect that BIO socket is created "exclusive". In fact one can wonder if they should be created "exclusive" unconditionally on Windows, as reflection of fact that it's "exclusive" on Unix... Thoughts?

Reply:
So they are saying that another process from another user account for instance
can intercept new connections with SO_REUSEADDR, unless we use
SO_EXCLUSIVADDRUSE?

Reply:
> So they are saying

Originally it was the case, yes. By now they have some improvements, like bound port can be hijacked only from same account...